cruise-control/src/main/java/com/linkedin/kafka/cruisecontrol/servlet/UserTaskManager.java

```diff
@@ -405,22 +405,31 @@ public synchronized UserTaskInfo markTaskExecutionBegan(String uuid) {
 
     if (_uuidToActiveUserTaskInfoMap.containsKey(userTaskId)) {
       _inExecutionUserTaskInfo = _uuidToActiveUserTaskInfoMap.remove(userTaskId).setState(TaskState.IN_EXECUTION);
-      // Normally a user task's operation result is logged when the task's state is transferred from ACTIVE to COMPLETED_WITH_ERROR.
-      // If the user task's state is transferred from ACTIVE directly to IN_EXECUTION, need to log the task's operation result here.
-      _inExecutionUserTaskInfo.logOperation();
     }
 
     return _inExecutionUserTaskInfo;
   }
 
+  /** Normally a user task's operation result is logged when the task's state is transferred from ACTIVE to COMPLETED or
+   * COMPLETED_WITH_ERROR. If the user task's state is transferred from ACTIVE directly to IN_EXECUTION,
+   * need to log the task's operation result.
+   */
+  public synchronized void logInExecutionTask() {
+    if (_inExecutionUserTaskInfo != null) {
+      _inExecutionUserTaskInfo.logOperation();
```

Comment:
I feel it's a bit hard to understand it at the first glance. Now I get it. Maybe we should call it logOperation or logTaskToBeExecuted, as at this moment the execution is not started yet.

And a nit: the first line of the javadoc comment is usually empty. Start the description from the second line.

Reply:
Changed the comment. Changed the method to "logInExecutionTaskOperation". We can consider this task is in execution because _hasOngoingExecution is already set True at this moment.